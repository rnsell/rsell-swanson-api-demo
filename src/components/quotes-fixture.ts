export const quoteFixture = {
  quoteText: "Go Beyond Plus Ultra!!",
  votes: 5,
  onUpVoteClick: jest.fn(() => void 0),
  onDownVoteClick: jest.fn(() => void 0),
  upVoteSelected: false,
  downVoteSelected: false,
};
