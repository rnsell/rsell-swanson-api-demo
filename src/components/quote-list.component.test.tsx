import React from "react";
import { mount } from "enzyme";
import { quoteFixture } from "./quotes-fixture";
import { QuoteList, StyledSkeleton } from "./quote-list.component";

describe("quote-list.component.tsx", () => {
  let component: any;
  const componentProps = {
    listOfQuotes: [quoteFixture],
    totalNumberOfQuotes: [quoteFixture].length,
  };

  beforeEach(() => {
    component = mount(<QuoteList {...componentProps} />);
  });

  it("Should match snapshot for not loading", () => {
    expect(component).toMatchSnapshot();
  });

  it("Should match snapshot for loading", () => {
    component.setProps({ isLoading: true });
    expect(component.find(StyledSkeleton)).toBeTruthy();

    expect(component).toMatchSnapshot();
  });
});
