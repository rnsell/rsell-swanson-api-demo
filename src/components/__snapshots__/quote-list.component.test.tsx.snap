// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`quote-list.component.tsx Should match snapshot for loading 1`] = `
<QuoteList
  isLoading={true}
  listOfQuotes={
    Array [
      Object {
        "downVoteSelected": false,
        "onDownVoteClick": [MockFunction],
        "onUpVoteClick": [MockFunction],
        "quoteText": "Go Beyond Plus Ultra!!",
        "upVoteSelected": false,
        "votes": 5,
      },
    ]
  }
  totalNumberOfQuotes={1}
>
  <List
    bordered={true}
    data-testid="quote-list"
    dataSource={
      Array [
        true,
      ]
    }
    renderItem={[Function]}
  >
    <div
      className="ant-list ant-list-split ant-list-bordered"
      data-testid="quote-list"
    >
      <Spin
        size="default"
        spinning={false}
        wrapperClassName=""
      >
        <div
          className="ant-spin-nested-loading"
        >
          <div
            className="ant-spin-container"
            key="container"
          >
            <ul
              className="ant-list-items"
            >
              <Row>
                <div
                  className="ant-row"
                  style={
                    Object {
                      "marginBottom": undefined,
                      "marginLeft": undefined,
                      "marginRight": undefined,
                      "marginTop": undefined,
                    }
                  }
                >
                  <Styled(Skeleton)
                    active={true}
                    paragraph={false}
                  >
                    <Skeleton
                      active={true}
                      avatar={false}
                      className="sc-jSgupP grydea"
                      paragraph={false}
                      title={true}
                    >
                      <div
                        className="ant-skeleton ant-skeleton-active sc-jSgupP grydea"
                      >
                        <div
                          className="ant-skeleton-content"
                        >
                          <Title
                            prefixCls="ant-skeleton-title"
                          >
                            <h3
                              className="ant-skeleton-title"
                              style={
                                Object {
                                  "width": undefined,
                                }
                              }
                            />
                          </Title>
                        </div>
                      </div>
                    </Skeleton>
                  </Styled(Skeleton)>
                </div>
              </Row>
            </ul>
          </div>
        </div>
      </Spin>
    </div>
  </List>
</QuoteList>
`;

exports[`quote-list.component.tsx Should match snapshot for not loading 1`] = `
<QuoteList
  listOfQuotes={
    Array [
      Object {
        "downVoteSelected": false,
        "onDownVoteClick": [MockFunction],
        "onUpVoteClick": [MockFunction],
        "quoteText": "Go Beyond Plus Ultra!!",
        "upVoteSelected": false,
        "votes": 5,
      },
    ]
  }
  totalNumberOfQuotes={1}
>
  <List
    bordered={true}
    data-testid="quote-list"
    dataSource={
      Array [
        Object {
          "downVoteSelected": false,
          "onDownVoteClick": [MockFunction],
          "onUpVoteClick": [MockFunction],
          "quoteText": "Go Beyond Plus Ultra!!",
          "upVoteSelected": false,
          "votes": 5,
        },
      ]
    }
    renderItem={[Function]}
  >
    <div
      className="ant-list ant-list-split ant-list-bordered"
      data-testid="quote-list"
    >
      <Spin
        size="default"
        spinning={false}
        wrapperClassName=""
      >
        <div
          className="ant-spin-nested-loading"
        >
          <div
            className="ant-spin-container"
            key="container"
          >
            <ul
              className="ant-list-items"
            >
              <QuoteComponent
                downVoteSelected={false}
                onDownVoteClick={[MockFunction]}
                onUpVoteClick={[MockFunction]}
                quoteText="Go Beyond Plus Ultra!!"
                upVoteSelected={false}
                votes={5}
              >
                <Item
                  data-testid="quote__list-item"
                >
                  <li
                    className="ant-list-item"
                    data-testid="quote__list-item"
                  >
                    <styled.div
                      data-testid="quote__row-button-container"
                    >
                      <div
                        className="sc-bdfBwQ beCqgx"
                        data-testid="quote__row-button-container"
                      >
                        <styled.div
                          data-testid="quote__button-container"
                        >
                          <div
                            className="sc-gsTCUz cPmiuJ"
                            data-testid="quote__button-container"
                          >
                            <Button
                              aria-label="Increase Vote"
                              data-testid="quote__upvote-button"
                              icon={<ForwardRef(LikeTwoTone) />}
                              onClick={[MockFunction]}
                              type="default"
                            >
                              <Wave>
                                <button
                                  aria-label="Increase Vote"
                                  className="ant-btn ant-btn-default ant-btn-icon-only"
                                  data-testid="quote__upvote-button"
                                  onClick={[Function]}
                                  type="button"
                                >
                                  <ForwardRef(LikeTwoTone)>
                                    <AntdIcon
                                      icon={
                                        Object {
                                          "icon": [Function],
                                          "name": "like",
                                          "theme": "twotone",
                                        }
                                      }
                                    >
                                      <span
                                        aria-label="like"
                                        className="anticon anticon-like"
                                        role="img"
                                      >
                                        <IconReact
                                          icon={
                                            Object {
                                              "icon": [Function],
                                              "name": "like",
                                              "theme": "twotone",
                                            }
                                          }
                                        >
                                          <svg
                                            aria-hidden="true"
                                            data-icon="like"
                                            fill="currentColor"
                                            focusable="false"
                                            height="1em"
                                            key="svg-like"
                                            viewBox="64 64 896 896"
                                            width="1em"
                                          >
                                            <path
                                              d="M273 495.9v428l.3-428zm538.2-88.3H496.8l9.6-198.4c.6-11.9-4.7-23.1-14.6-30.5-6.1-4.5-13.6-6.8-21.1-6.7-19.6.1-36.9 13.4-42.2 32.3-37.1 134.4-64.9 235.2-83.5 302.5V852h399.4a56.85 56.85 0 0033.6-51.8c0-9.7-2.3-18.9-6.9-27.3l-13.9-25.4 21.9-19a56.76 56.76 0 0019.6-43c0-9.7-2.3-18.9-6.9-27.3l-13.9-25.4 21.9-19a56.76 56.76 0 0019.6-43c0-9.7-2.3-18.9-6.9-27.3l-14-25.5 21.9-19a56.76 56.76 0 0019.6-43c0-19.1-11-37.5-28.8-48.4z"
                                              fill="#e6f7ff"
                                              key="svg-like-svg-0"
                                            />
                                            <path
                                              d="M112 528v364c0 17.7 14.3 32 32 32h65V496h-65c-17.7 0-32 14.3-32 32zm773.9 5.7c16.8-22.2 26.1-49.4 26.1-77.7 0-44.9-25.1-87.5-65.5-111a67.67 67.67 0 00-34.3-9.3H572.3l6-122.9c1.5-29.7-9-57.9-29.5-79.4a106.4 106.4 0 00-77.9-33.4c-52 0-98 35-111.8 85.1l-85.8 310.8-.3 428h472.1c9.3 0 18.2-1.8 26.5-5.4 47.6-20.3 78.3-66.8 78.3-118.4 0-12.6-1.8-25-5.4-37 16.8-22.2 26.1-49.4 26.1-77.7 0-12.6-1.8-25-5.4-37 16.8-22.2 26.1-49.4 26.1-77.7 0-12.6-1.8-25-5.4-37zM820.4 499l-21.9 19 14 25.5a56.2 56.2 0 016.9 27.3c0 16.5-7.1 32.2-19.6 43l-21.9 19 13.9 25.4a56.2 56.2 0 016.9 27.3c0 16.5-7.1 32.2-19.6 43l-21.9 19 13.9 25.4a56.2 56.2 0 016.9 27.3c0 22.4-13.2 42.6-33.6 51.8H345V506.8c18.6-67.2 46.4-168 83.5-302.5a44.28 44.28 0 0142.2-32.3c7.5-.1 15 2.2 21.1 6.7 9.9 7.4 15.2 18.6 14.6 30.5l-9.6 198.4h314.4C829 418.5 840 436.9 840 456c0 16.5-7.1 32.2-19.6 43z"
                                              fill="#1890ff"
                                              key="svg-like-svg-1"
                                            />
                                          </svg>
                                        </IconReact>
                                      </span>
                                    </AntdIcon>
                                  </ForwardRef(LikeTwoTone)>
                                </button>
                              </Wave>
                            </Button>
                            <styled.div
                              aria-label="Total Votes"
                              data-testid="quote__quote-counter-container"
                            >
                              <div
                                aria-label="Total Votes"
                                className="sc-hKgILt lkUzfm"
                                data-testid="quote__quote-counter-container"
                              >
                                <styled.div
                                  data-testid="quote__quote-counter-justification-container"
                                >
                                  <div
                                    className="sc-eCssSg fhiQTM"
                                    data-testid="quote__quote-counter-justification-container"
                                  >
                                    5
                                  </div>
                                </styled.div>
                              </div>
                            </styled.div>
                            <Button
                              aria-label="Decrease Vote"
                              data-testid="quote__downvote-button"
                              icon={<ForwardRef(DislikeTwoTone) />}
                              onClick={[MockFunction]}
                              type="default"
                            >
                              <Wave>
                                <button
                                  aria-label="Decrease Vote"
                                  className="ant-btn ant-btn-default ant-btn-icon-only"
                                  data-testid="quote__downvote-button"
                                  onClick={[Function]}
                                  type="button"
                                >
                                  <ForwardRef(DislikeTwoTone)>
                                    <AntdIcon
                                      icon={
                                        Object {
                                          "icon": [Function],
                                          "name": "dislike",
                                          "theme": "twotone",
                                        }
                                      }
                                    >
                                      <span
                                        aria-label="dislike"
                                        className="anticon anticon-dislike"
                                        role="img"
                                      >
                                        <IconReact
                                          icon={
                                            Object {
                                              "icon": [Function],
                                              "name": "dislike",
                                              "theme": "twotone",
                                            }
                                          }
                                        >
                                          <svg
                                            aria-hidden="true"
                                            data-icon="dislike"
                                            fill="currentColor"
                                            focusable="false"
                                            height="1em"
                                            key="svg-dislike"
                                            viewBox="64 64 896 896"
                                            width="1em"
                                          >
                                            <path
                                              d="M273 100.1v428h.3l-.3-428zM820.4 525l-21.9-19 14-25.5a56.2 56.2 0 006.9-27.3c0-16.5-7.1-32.2-19.6-43l-21.9-19 13.9-25.4a56.2 56.2 0 006.9-27.3c0-16.5-7.1-32.2-19.6-43l-21.9-19 13.9-25.4a56.2 56.2 0 006.9-27.3c0-22.4-13.2-42.6-33.6-51.8H345v345.2c18.6 67.2 46.4 168 83.5 302.5a44.28 44.28 0 0042.2 32.3c7.5.1 15-2.2 21.1-6.7 9.9-7.4 15.2-18.6 14.6-30.5l-9.6-198.4h314.4C829 605.5 840 587.1 840 568c0-16.5-7.1-32.2-19.6-43z"
                                              fill="#e6f7ff"
                                              key="svg-dislike-svg-0"
                                            />
                                            <path
                                              d="M112 132v364c0 17.7 14.3 32 32 32h65V100h-65c-17.7 0-32 14.3-32 32zm773.9 358.3c3.6-12 5.4-24.4 5.4-37 0-28.3-9.3-55.5-26.1-77.7 3.6-12 5.4-24.4 5.4-37 0-28.3-9.3-55.5-26.1-77.7 3.6-12 5.4-24.4 5.4-37 0-51.6-30.7-98.1-78.3-118.4a66.1 66.1 0 00-26.5-5.4H273l.3 428 85.8 310.8C372.9 889 418.9 924 470.9 924c29.7 0 57.4-11.8 77.9-33.4 20.5-21.5 31-49.7 29.5-79.4l-6-122.9h239.9c12.1 0 23.9-3.2 34.3-9.3 40.4-23.5 65.5-66.1 65.5-111 0-28.3-9.3-55.5-26.1-77.7zm-74.7 126.1H496.8l9.6 198.4c.6 11.9-4.7 23.1-14.6 30.5-6.1 4.5-13.6 6.8-21.1 6.7a44.28 44.28 0 01-42.2-32.3c-37.1-134.4-64.9-235.2-83.5-302.5V172h399.4a56.85 56.85 0 0133.6 51.8c0 9.7-2.3 18.9-6.9 27.3l-13.9 25.4 21.9 19a56.76 56.76 0 0119.6 43c0 9.7-2.3 18.9-6.9 27.3l-13.9 25.4 21.9 19a56.76 56.76 0 0119.6 43c0 9.7-2.3 18.9-6.9 27.3l-14 25.5 21.9 19a56.76 56.76 0 0119.6 43c0 19.1-11 37.5-28.8 48.4z"
                                              fill="#1890ff"
                                              key="svg-dislike-svg-1"
                                            />
                                          </svg>
                                        </IconReact>
                                      </span>
                                    </AntdIcon>
                                  </ForwardRef(DislikeTwoTone)>
                                </button>
                              </Wave>
                            </Button>
                          </div>
                        </styled.div>
                        <styled.div
                          aria-label="Quote Text"
                          data-testid="quote__quote-text"
                        >
                          <div
                            aria-label="Quote Text"
                            className="sc-dlfnbm"
                            data-testid="quote__quote-text"
                          >
                            Go Beyond Plus Ultra!!
                          </div>
                        </styled.div>
                      </div>
                    </styled.div>
                  </li>
                </Item>
              </QuoteComponent>
            </ul>
          </div>
        </div>
      </Spin>
    </div>
  </List>
</QuoteList>
`;
